### DISCUSSION TITLE
Composing PNG Image over an Animated GIF File using Magick.NET?

### DISCUSSION THREAD

**Discussion by pariswhitney** [dlemstra/Magick.NET#945] [Help]
How is it possible with Magick.net in C# (not with CLI) to put an **alpha.png** image over an single **animated.gif** file and save this composing again as an single **animated gif**? 

![alpha_image](https://user-images.githubusercontent.com/85218283/120470164-87c74080-c3a3-11eb-9756-25cf8a54a7dc.png)

![animated_background](https://user-images.githubusercontent.com/85218283/120470170-885fd700-c3a3-11eb-9691-b04f81343eb2.gif)

![final_composing](https://user-images.githubusercontent.com/85218283/120470171-88f86d80-c3a3-11eb-8ce0-7aefba866664.gif)

**Comment by dlemstra** [comment]
You can do that with the following sample:

```C#
using (var images = new MagickImageCollection("animated.gif"))
{
    images.Coalesce();

    using (var alpha = new MagickImage("alpha.png"))
    {
        foreach (var image in images)
        {
            image.Composite(alpha, CompositeOperator.Over);
        }
    }

    images.Optimize();

    images.Write("output.gif");
}
```
